src/RpcClient.ts

```diff
@@ -354,4 +382,111 @@ export class RpcClient {
     }
   }
+  /**
+   * Mint a compressed NFT.
+   * @returns {Promise<MintCNFTResponse>}
+   * @throws {Error}
+   */
+  async mintCNFT(
+    request: MintCNFTRequest
+  ): Promise<MintCNFTResult | undefined> {
+    try {
+      const ownerKeypair = loadWallet(request.ownerKeypair);
+      function delay(ms: number) {
+        return new Promise((resolve) => setTimeout(resolve, ms));
+      }
+      const defaultMetadataArgs: MetadataArgs = {
+        ...request.metadataArgs,
+        primarySaleHappened: false,
+        isMutable: false,
+        editionNonce: 0,
+        tokenStandard: null,
+        uses: null,
+        tokenProgramVersion: null,
+      };
+      let assetId = "";
+      let treeWallet = request.treeKeypair
+        ? loadWallet(request.treeKeypair)
+        : Keypair.generate();
+      if (!request.treeKeypair) {
+        await initTree(this.connection, ownerKeypair, treeWallet);
+      }
+      let response: any | undefined;
+
+      if (!request.collection) {
+        const signature = await mintCompressedNFT(
+          this.connection,
+          defaultMetadataArgs,
+          request.ownerKeypair,
+          treeWallet
+        );
+        await delay(4000);
+        const { id } = await getAssetID(
+          this.connection.rpcEndpoint,
+          treeWallet.publicKey
+        );
+        response = {
+          assetId: id,
+          creator: request.ownerKeypair,
+          treeId: treeWallet.publicKey.toString(),
+          treeKeypair: treeWallet.secretKey,
+          signature: signature,
+        };
+        assetId = id;
+      } else {
+        let collectionMint = request.collectionMint;
+        if (!collectionMint) {
+          const collectionInfo = await initCollection(
+            this.connection,
+            ownerKeypair,
+            request.metadataArgs.name,
+            request.metadataArgs.symbol,
+            request.metadataArgs.uri
+          );
+          collectionMint = collectionInfo.collectionMint;
+        }
+
+        const { collectionMetadataAccount, collectionMasterEditionAccount } =
+          await existingCollection(collectionMint);
+        const signature = await mintCollectionCompressedNft(
+          this.connection,
+          defaultMetadataArgs,
+          request.ownerKeypair,
+          treeWallet,
+          collectionMint,
+          collectionMetadataAccount,
+          collectionMasterEditionAccount
+        );
+        await delay(4000);
+        // Get Merkle tree data after minting the NFT
+        const { id } = await getAssetID(
+          this.connection.rpcEndpoint,
+          treeWallet.publicKey
+        );
+        assetId = id;
+        response = {
+          assetId: id,
+          creator: request.ownerKeypair,
+          treeId: treeWallet.publicKey.toString(),
+          treeKeypair: treeWallet.secretKey,
+          signature: signature,
+        };
+      }
+
+      let DAS;
+      if (request.dasPolling == true) {
```

Comment:
this isn't polling, you only called it once. polling is when you repeatedly call until the response arrives.